[MENFORCER-398] show rules processed

raupachz opened the pull request:
Hi there,

this fixes [MENFORCER-398](https://issues.apache.org/jira/browse/MENFORCER-398). It is a simple one line change. Turning on DEBUG already logs what rules are processed. Since most people probably run Maven in default settings they won't see the rules processed. Changing the log level from DEBUG to INFO allows everyone to see the rule processed.

This is an up-for-grabs issue. Feedback welcome.

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MENFORCER) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MENFORCER-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MENFORCER-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).



 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

slawekjaranowski commented on maven-enforcer-plugin/src/main/java/org/apache/maven/plugins/enforcer/EnforceMojo.java:
```diff
@@ -187,7 +187,7 @@ public void execute()
                 // logging purposes
                 currentRule = rule.getClass().getName();
-                log.debug( "Executing rule: " + currentRule );
+                log.info( "Executing rule: " + currentRule );
```
Rule is executed in line 199, we will have such message even when rule will not be executed in case of cache.

Is it what we want?

slawekjaranowski reviewed (commented):
it will be appreciated to add simple assertion in one of existing it test, to see how it is look

slawekjaranowski commented:
Please also put issue number in square brackets in commit message and PR title

raupachz commented:
Thanks for the feedback @slawekjaranowski !

I updated the pull request accordingly.
- the commit message now has the issue number in square brackets and I added the PR title
- moved the log.info message right bevor the `execute` method. (Sorry, I was not aware of cache)
- Added an assertion in `testEnforceMojo`

slawekjaranowski commented:
@raupachz thanks

jira-importer commented:
Resolve #515
